src/libraries/PoolStateLibrary.sol

```diff
@@ -0,0 +1,350 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.21;
+
+import {PoolId} from "../types/PoolId.sol";
+import {IPoolManager} from "../interfaces/IPoolManager.sol";
+import {Position} from "./Position.sol";
+
+library PoolStateLibrary {
+    // forge inspect src/PoolManager.sol:PoolManager storage --pretty
+    // | Name                  | Type                                    | Slot | Offset | Bytes | Contract                        |
+    // |-----------------------|-----------------------------------------|------|--------|-------|---------------------------------|
+    // | pools                 | mapping(PoolId => struct Pool.State)    | 6    | 0      | 32    | src/PoolManager.sol:PoolManager |
+    uint256 public constant POOLS_SLOT = 6;
+
+    // index of feeGrowthGlobal0X128 in Pool.State
+    uint256 public constant FEE_GROWTH_GLOBAL0_OFFSET = 1;
+    // index of feeGrowthGlobal1X128 in Pool.State
+    uint256 public constant FEE_GROWTH_GLOBAL1_OFFSET = 2;
+
+    // index of liquidity in Pool.State
+    uint256 public constant LIQUIDITY_OFFSET = 3;
+
+    // index of TicksInfo mapping in Pool.State
+    uint256 public constant TICK_INFO_OFFSET = 4;
```

Comment:
this and `POSITION_INFO_OFFSET` feel like they should be renamed `TICKS_OFFSET` and `POSITIONS_OFFSET` as thats the names of the mapping